### DISCUSSION TITLE
Stateful Conformal Predictors

### DISCUSSION THREAD

**Discussion by Schinkikami** [awslabs/fortuna#91] [Ideas]
Hey, I just found your library and I am really excited, that you offer a platform that implements the CP algorithms. So far most of what I have used was hand-implemented. 

However, why are the CP classes (e.g. AdaptivePredictionConformalClassifier) not stateful? To me, it seems logical to initialize the CP algorithm with the calibration probabilities and target variables so that during initialization the quantile is computed. Then at a later point, this class (and it's precomputed quantile) is reused to compute conformal sets for given inputs during deployment. 

If for example such a system should be used in a live environment, the production of conformal sets should be able to run in a fast and responsive loop. However, currently, you would need to recompute the conformal quantile each time from the whole (identical) calibration dataset, to produce conformal sets.

**Comment by gianlucadetommaso** [comment]
Hi!
Thanks for the message. You raise a good point. You can currently do what you want by first computing the quantiles using `AdaptivePredictionConformalClassifier.quantile`, and then passing the computed quantiles to `AdaptivePredictionConformalClassifier.conformal_set`. An example:

```
from fortuna.conformal import AdaptivePredictionConformalClassifier
conformal_classifier = AdaptivePredictionConformalClassifier()
quantiles = conformal_classifier.quantile(...)
conformal_sets = conformal_classifier.conformal_set(...)
```

Your solution is also possible, but I found this one a bit lighter.
